src/main/java/org/jenkinsci/plugins/p4/groovy/P4Groovy.java

```diff
@@ -100,7 +103,7 @@ public String getUserName() throws Exception {
 	}
 
 	private IOptionsServer getConnection() throws IOException {
-		ClientHelper p4 = new ClientHelper(Jenkins.getActiveInstance(), credential, listener, workspace);
+		ClientHelper p4 = new ClientHelper(p4Credential, listener, workspace);
```

Comment:
I think this is the main fix.  Not using the active Instance and looking up the credential from the string ID.